### PR TITLE
fix: trim peer str

### PR DESCRIPTION
**Pull request by mindfocus**
### Motivation:

trim peerstr，使其更健壮？

### Modification:

trim peerstr，使其更健壮？

### Result:

Fixes #<GitHub issue number>.

If there is no issue then describe the changes introduced by this PR.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved configuration parsing by trimming spaces in configuration strings.

- **Tests**
  - Added a new test to verify configuration parsing with spaces.

### CODE REVIEW

File: jraft-core/src/main/java/com/alipay/sofa/jraft/conf/Configuration.java [hunk 1]
```diff
@@ -297,7 +297,7 @@ public boolean parse(final String conf) {
                 peerStr = peerStr.substring(0, index);
                 isLearner = true;
             }
-            if (peer.parse(peerStr)) {
+            if (peer.parse(StringUtils.trim(peerStr))) {
```
[the remark this answers is not in the document]
Comment (fengjiachun) [reply jraft-core/src/main/java/com/alipay/sofa/jraft/conf/Configuration.java:300]: Please help to write a unit test

**Review by fengjiachun — APPROVED** [review]
LGTM, thank you!

**Review by caicancai — APPROVED** [review]
LGTM

**Comment by fengjiachun** [comment]
CI failed @mindfocus

**Comment by mindfocus** [comment]
@fengjiachun fixed.

**Pull request merged**